src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_ChangeNamespace.cs

```diff
@@ -0,0 +1,1858 @@
+        {
+            var defaultNamespace = "A";
+            var documentPath = CreateDocumentFilePath(new[] { "B", "C" }, "File1.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+using System;
+
+class [||]Class1
+{{
+}}
+</Document>
+    </Project>
+</Workspace>";
+
+            var expectedSourceOriginal =
+@"using System;
+
+namespace A.B.C
+{
+    class Class1
+    {
+    }
+}";
+            await TestChangeNamespaceAsync(code, expectedSourceOriginal);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task ChangeFromGlobalNamespace_SingleDocumentLocalRef()
+        {
+            var defaultNamespace = "A";
+            var documentPath = CreateDocumentFilePath(new[] { "B", "C" }, "File1.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath.folder}"" FilePath=""{documentPath.filePath}""> 
+delegate void [||]D1;
+
+interface Class1
+{{
+    void M1();
+}}
+
+class Class2 : Class1
+{{
+    D1 d;  
+
+    void Class1.M1() {{ }}
+}}</Document>
+    </Project>
+</Workspace>";
+
+            var expectedSourceOriginal =
+@"namespace A.B.C
+{
+    delegate void D1;
+
+    interface Class1
+    {
+        void M1();
+    }
+
+    class Class2 : Class1
+    {
+        D1 d;
+
+        void Class1.M1() { }
+    }
+}";
+            await TestChangeNamespaceAsync(code, expectedSourceOriginal);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task ChangeFromGlobalNamespace_WithReferencesInOtherDocument()
+        {
+            var defaultNamespace = "A";
+
+            var documentPath1 = CreateDocumentFilePath(new[] { "B", "C" }, "File1.cs");
+            var documentPath2 = CreateDocumentFilePath(Array.Empty<string>(), "File2.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath1.folder}"" FilePath=""{documentPath1.filePath}""> 
+class [||]Class1 
+{{ 
+}}
+    
+class Class2 
+{{ 
+}}</Document>
+<Document Folders=""{documentPath2.folder}"" FilePath=""{documentPath2.filePath}""> 
+namespace Foo
+{{
+    class RefClass
+    {{
+        private Class1 c1;
+
+        void M1()
+        {{
+            Class2 c2 = null;
+        }}
+    }}
+}}</Document>
+    </Project>
+</Workspace>";
+
+            var expectedSourceOriginal =
+@"namespace A.B.C
+{
+    class Class1
```

Comment:
not specifically related to sync namespace. but the change namespace thing, how will it deal with conflict? it doesn't care? (leave things broken), or still do but show some warning somewhere? or bail out on such case?

asking since "global" case seems could introduce conflict quite easily. (but since it require explicit user action to apply, I dont think it matter much, just curious)

Reply:
It doesn't try to detect any potential conflicts, and will leave code broken if there's any. This is mostly for perf reason, plus making it right is difficult. We do show a warning for this code action that the refactoring might break your code (or worse, change the semantic without errors)